### PR TITLE
Check the managed cluster for the multiline tests

### PR DESCRIPTION
**Pull request by dhaiducek**
When run in the canaries, this policy is getting deployed to the Managed cluster, but all the checks were against the hub cluster.

Also does some refactoring to loop over the hub and managed clusters.

ref: https://github.com/stolostron/backlog/issues/27232

### CODE REVIEW

**Comment by dhaiducek** [comment]
CI here is passing. This PR is ready for review.

File: test/integration/policy_multiline_templatization_test.go [hunk 1]
```diff
@@ -68,43 +67,31 @@ var _ = Describe(
...
-					if err != nil {
-						return ""
-					}
-
-					return configMap.Data["extraData"]
-				},
-				defaultTimeoutSeconds*2,
-				1,
-			).Should(Equal("exists!"))
+		It("The ConfigMaps on the managed cluster should be patched with the correct data", func() {
```
Comment (mprahl) [inline test/integration/policy_multiline_templatization_test.go:70]: Should this also verify the Hub results as well?

Comment (dhaiducek) [reply test/integration/policy_multiline_templatization_test.go:70]: I don't think so because there's no guarantee that the hub is self-managed.

Comment (dhaiducek) [reply test/integration/policy_multiline_templatization_test.go]: It's an interesting point. I'll add in the check, but it does make me wonder whether we ever test a non-self-managed hub with a managed cluster. I guess not...

**Review by mprahl — APPROVED** [review]

**Pull request merged**